File: 2019 czerwiec/python/3.py
with open("pierwsze.txt", "r") as f:
    liczby = list(map(int, f.read().strip().split('\n')))

def w(n):
    if n < 10:
        return n
    
    ans = 0
    while n:
        ans += n % 10
        n //= 10
    return w(ans)

ans = 0
with open("wyniki4_3.txt", "w") as f:
    for n in liczby:
        if w(n) == 1:
            ans += 1
    f.write(str(ans))

File: 2019 czerwiec/c++/3.cpp
#include <iostream>
#include <fstream>
using namespace std;

int waga(int n)
{
    if (n < 10)
        return n;
    int ans = 0;
    while (n)
    {
        ans += n % 10;
        n /= 10;
    }
    return waga(ans);
}

int main()
{
    ifstream in("pierwsze.txt");
    ofstream out("wyniki4_3.txt");

    int ans = 0;
    for (int i = 1; i <= 200; i++)
    {
        int a;
        in >> a;
        if (waga(a) == 1)
            ans++;
    }
    out << ans;
}